haystack/components/routers/file_type_router.py

```diff
@@ -86,7 +86,7 @@ def run(self, sources: List[Union[str, Path, ByteStream]]) -> Dict[str, List[Uni
             if isinstance(source, Path):
                 mime_type = self._get_mime_type(source)
             elif isinstance(source, ByteStream):
-                mime_type = source.meta.get("content_type", None)
+                mime_type = source.resolved_mime_type
```

Comment:
I think the only change should be
```suggestion
                mime_type = source.mime_type
```

Reply:
Hey @masci @wochinge and I went back and forth on this one in the background for several rounds before deciding on this approach. 

We can't employ this simple solution without breaking some existing components i.e. ContentFetcher that uses metadata of the bytestream to set content_type. We have diversion between local files that set mime_type directly and web based fetches that set content type. And now this property resolves the mime_type regardless of where the data comes from. That's the idea.

Reply:
It's not really breaking, it's more that those components are bugged no? I think by supporting the buggy behaviour we pile up tech debt and make the code more complex, is it really worth it?

Reply:
Yeah, i can do that as well. We can find all the references of content_type and replace it, in one PR. Wdyt @wochinge ?

Reply:
Yeah this will work, we have to change a few things in various tests, in link content fetcher but the change won't have widespread impact on users. Thanks @masci

Reply:
Agree that just relying on `.mime_type` seems simpler - hard for me to overlook the requirements of these other components though so I don't think i'm a good judge here